src/main/java/seedu/momentum/commons/core/StatisticTimeframe.java

```diff
@@ -0,0 +1,72 @@
+package seedu.momentum.commons.core;
+
+import java.io.Serializable;
+import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+
+/**
+ * Timeframe of statistics tracked.
+ */
+public class StatisticTimeframe implements Serializable {
+
+    public static final String MESSAGE_CONSTRAINTS =
+        "Timeframe should be either 'daily', 'weekly', or 'monthly'.";
+
+    private Timeframe timeframe;
+
+    /**
+     * Empty constructor required by Jackson.
+     */
+    public StatisticTimeframe() {
+    }
+
+    /**
+     * Constructs a {@code StatisticTimeframe} with the specified timeframe.
+     */
+    public StatisticTimeframe(Timeframe timeframe) {
+        this.timeframe = timeframe;
+    }
+
+    /**
+     * Converts the timeframe to a {@code ChronoUnit}.
+     */
+    public ChronoUnit toChronoUnit() {
+        return timeframe == Timeframe.DAILY
+            ? ChronoUnit.DAYS
+            : timeframe == Timeframe.WEEKLY
+            ? ChronoUnit.WEEKS
+            : ChronoUnit.MONTHS;
+    }
```

Comment:
Implement the conversion from enum Timeframe to ChronoUnit in the enum class instead then call the method here?